### PR TITLE
Implement pagination in the search client

### PR DESCRIPTION
**Pull request by craiga**
Pagination in the search client

### CODE REVIEW

File: medicines/search-client/src/lib.rs [hunk 1]
```diff
@@ -166,43 +192,70 @@ impl CreateIndexEntry for AzureSearchClient {
...
+    match (result_count, offset) {
+        (Some(result_count), Some(offset)) => Ok(request_builder
+            .query(&[
+                ("$top", result_count.to_string()),
+                ("$skip", offset.to_string()),
+            ])
+            .build()?),
+        (None, None) => Ok(request_builder.build()?),
+        _ => Err(anyhow::anyhow!(
+            "If one of result_count or offset is set, the other must be set as well."
```
Comment (rjkerrison) [inline medicines/search-client/src/lib.rs:256]: could they be unified into a `Pagination` type, then we pass in just `Option<Pagination>`? even `Option<(u32, u32>) should work

File: medicines/search-client/src/lib.rs [hunk 2]
```diff
@@ -166,43 +192,70 @@ impl CreateIndexEntry for AzureSearchClient {
...
         .error_for_status()?
         .json::<IndexResults>()
         .await
+        .map_err(|e| anyhow::anyhow!(e))
 }
 
 fn build_search(
     search_term: &str,
+    result_count: Option<u32>,
+    offset: Option<u32>,
```
Comment (rjkerrison) [inline medicines/search-client/src/lib.rs:222]: could these be `i32` instead? I don't know what a negative integer would be useful for

Comment (craiga) [reply medicines/search-client/src/lib.rs:222]: They could be `i32`, but that would allow negative result count and offsets, which don't make any sense.

Comment (rjkerrison) [reply medicines/search-client/src/lib.rs:222]: right, think I just muddled up which integer is which, don't mind me

**Pull request merged**